CalorieTracker/CalorieTracker/Model/CoreDataStack.swift

```diff
@@ -0,0 +1,44 @@
+//
+//  CoreDataStack.swift
+//  CalorieTracker
+//
+//  Created by Craig Swanson on 2/23/20.
+//  Copyright © 2020 Craig Swanson. All rights reserved.
+//
+
+import Foundation
+import CoreData
+
+class CoreDataStack {
+    
+    static let shared = CoreDataStack()
+    
+    lazy var container: NSPersistentContainer = {
+        let container = NSPersistentContainer(name: "CalorieTracker")
+        container.loadPersistentStores { _, error in
+            if let error = error {
+                fatalError("Failed to load persistent stores: \(error)")
+            }
+        }
+        container.viewContext.automaticallyMergesChangesFromParent = true
+        return container
+    }()
+    
+    var mainContext: NSManagedObjectContext {
+        return container.viewContext
+    }
+    
+    func save(context: NSManagedObjectContext =
+        CoreDataStack.shared.mainContext) throws {
+        var error: Error?
+        
+        context.performAndWait {
+            do {
+                try context.save()
```

Comment:
This is how I usually implement my save(context:). But wanted to let you know that there is an instance variable on the context named "hasChanged" which is a boolean value that indicates whether the context has uncommitted changes. 

Some developers check to see if there has been any changes on the context, and if so, then they call context.save().